perf/pgbench.py

```diff
@@ -0,0 +1,127 @@
+                            f"with the text '{command.result.stderr_text}'")
+        return [command.result.exit_status, command.result.stdout_text]
+
+    def get_pkg_manager(self):
+        for pkg_mgr in ['apt', 'dnf', 'yum']:
+            if self.sm.check_installed(pkg_mgr):
+                return pkg_mgr
+        return None
+
+    def install_packages(self, pkg_mgr):
+        packages = ["postgresql", "postgresql-contrib"]
+        for package in packages:
+            self.sm.install(package)
+
+        # rhel needs an extra pkg
+        if pkg_mgr in ['dnf', 'yum']:
+            dnf_pkgs = ["postgresql-server"]
+            for package in dnf_pkgs:
+                self.sm.install(package)
+
+            # restart postgres service
+            ManagerService = ServiceManager()
+            ManagerService.restart("postgresql")
+
+    def setUp(self):
+        '''
+        Install Pgbench
+        '''
+        # check root privileges ($USER should be able to run commands
+        # as another user without requiring a password)
+        if os.geteuid() != 0:
+            self.cancel("This script requires root privileges, "
+                        "Please try again with 'sudo' or as 'root'")
+
+        # check if pgbench is installed
+        if (self.sm.check_installed("pgbench")):
+            self.log.info("Pgbench is already installed")
+        else:
+            pkg_mgr = self.get_pkg_manager()
+            if pkg_mgr:
+                self.install_packages(pkg_mgr)
+            else:
+                self.cancel("Unsupported Linux distribution")
+
+        # add current user to db
+        output = self.run_cmd(f"sudo -u postgres createuser --superuser {os.getlogin()}", True)
+        if output[0] and "already exists" not in output[1]:
+            self.cancel(f"Could not create role {os.getlogin()}")
+
+    def test(self):
+
+        # Setup the params
+        scaling_factor = self.params.get("scaling_factor", default=100)
+        worker_threads = self.params.get("worker_threads", default=cpu.online_count())
+        db_clients = self.params.get("db_clients", default=cpu.online_count())
+        protocol = self.params.get("protocol", default="prepared")
+        benchmark_duration = self.params.get("benchmark_duration", default=0)
+        transaction_count = self.params.get("transaction_count", default=0)
+
+        # validation: pgbench cannot accept both transaction count and duration
+        if benchmark_duration > 0 and transaction_count > 0:
+            self.log.warn("Pgbench cannot accept both max transaction count and duration of the test, proceeding with specified time limit")
+        elif benchmark_duration == 0 and transaction_count == 0:
+            self.log.warn("Transaction count / duration of the test not specified, defaulting to time limit of 120 secs")
+            benchmark_duration = 120
+
+        # create the db
+        self.run_cmd("createdb pgbench", False)
```

Comment:
I think the steps to create and initalize the pgbench db are better done in setUp() rather than run_cmd()